### PR TITLE
Use Notification Hook

### PR DESCRIPTION
**Pull request by AugusteBa**

### CODE REVIEW

**Comment by AugusteBa** [comment]
@oriondean, again, cannot seem to add you as a reviewer, but I would greatly appreciate it if you had a look.

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (oriondean) [inline .vscode/settings.json]: Should this be comitted?

File: demo/src/common/LaunchConfiguration.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
```
Comment (oriondean) [inline demo/src/common/LaunchConfiguration.tsx]: Could `LaunchConfiguration` and `Demo` components be combined?

File: demo/src/pages/notification/Notification.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        notificationOptions: launchConfig.notificationOptions,
+        parentDocument: document,
+        shouldInheritCss: launchConfig.shouldInheritCss,
+        shouldInheritScripts: launchConfig.shouldInheritScripts,
+    };
+
+    return (
+        <div className={styles.container}>
+            <h1 className={styles.title}>useBounds</h1>
+            <div className={styles.description}>
```
Comment (oriondean) [inline demo/src/pages/notification/Notification.tsx]: This description needs updating

File: demo/src/pages/notification/Notification.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        message: null,
+        timeout: "never",
+        url: HTML_URL,
+    },
+    shouldInheritCss: true,
+    shouldInheritScripts: true,
+    shouldLoadJsxAfterLaunch: true,
+};
+
+const Notification: React.FC = () => {
```
Comment (oriondean) [inline demo/src/pages/notification/Notification.tsx]: There's an issue here if i try and populate a dismissed notification / closed child window

File: index.d.ts [hunk 5]
```diff
@@ -66,18 +85,28 @@ export const useChannelProvider: (
 };
 
 export enum enums {
```
Comment (oriondean) [inline index.d.ts]: Is this used?

File: src/useChildWindow.ts [hunk 6]
```diff
@@ -3,177 +3,178 @@ import { WindowOption } from "openfin/_v2/api/window/windowOption";
...
+        (nodes: HTMLCollectionOf<HTMLStyleElement | HTMLScriptElement>) => {
+            // tslint:disable-next-line: prefer-for-of
+            for (let i = 0; i < nodes.length; i++) {
+                injectNode(nodes[i]);
+            }
+        },
+        [injectNode],
+    );
+
+    const inheritScripts = useCallback(() => {
```
Comment (oriondean) [inline src/useChildWindow.ts]: Could we de-dupe this injectNodes based callback? It's used in a few places

File: .vscode/settings.json (hunk 1 above)
Comment (AugusteBa) [reply .vscode/settings.json]: No, it shouldn't, thanks!

File: demo/src/pages/notification/Notification.tsx (hunk 4 above)
Comment (AugusteBa) [reply demo/src/pages/notification/Notification.tsx]: Very good point. I now added onDissmiss, onClose overrides that reset all state variables.

File: index.d.ts (hunk 5 above)
Comment (AugusteBa) [reply index.d.ts]: Can be safely removed indeed.

File: demo/src/pages/notification/components/DemoActions.tsx [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+                onClick={() => notification.launch(notificationOptions)}
+                disabled={notification.state !== "INITIAL"}
+            >
+                Launch
+      </button>
+            <button
+                onClick={() =>
+                    notification.populate(<>{ReactHtmlParser(textAreaValue)}</>)
+                }
+                disabled={!notification.ref}
```
Comment (oriondean) [inline demo/src/pages/notification/components/DemoActions.tsx]: Can this be disabled when the notification is dismissed?

File: demo/tsconfig.json [hunk 8]
```diff
@@ -20,6 +20,6 @@
     "skipLibCheck": true
   },
   "include": [
-    "src"
+    "src", "../src/utils/helpers/inject.ts"
```
Comment (oriondean) [inline demo/tsconfig.json]: Why is this needed?

Comment (AugusteBa) [reply demo/tsconfig.json]: Must be `tslint fix` command that added this. It is not needed indeed. Thanks!

File: demo/src/pages/notification/components/DemoActions.tsx (hunk 7 above)
Comment (AugusteBa) [reply demo/src/pages/notification/components/DemoActions.tsx]: It is disabled as notification is dismissed as we just tested.
